File: test_api.py
import requests
import json
from pprint import pprint

id_comp = "CL"
year = 2022
url = f"https://api.football-data.org/v4/competitions/{
    id_comp}/teams"
url
params = {
    "season": year
}
headers = {
    'X-Auth-Token': '7ee3394168a140f98a511dd870124818'
}

response = requests.get(url, headers=headers, params=params)

#afficher l'url de la requête
print(response.url)

#afficher le code de statut de la requête
print(response.status_code)

File: test_api.r
library(httr)
library(jsonlite)
library(dplyr)

api_url <- 'https://api.football-data.org/v4/competitions/matches/'
headers <- add_headers('X-Auth-Token' = '7ee3394168a140f98a511dd870124818')

response <- GET(api_url, headers)
if (status_code(response) == 200) {
  data <- content(response, as = "text")
  data <- fromJSON(data)
  df <- data$matches
  print(head(df))
}
